src/commonmark/en/content/developer/web-api.md

```diff
@@ -16464,6 +16476,236 @@ based on data element
+</tr>
+<tr class="odd">
+<td>program</td>
+<td>Object containing the id of the program.</td>
+<td>"program" : {
+"id":"a3kGcGDCuk6"
+}</td>
+</tr>
+<tr class="even">
+<td>programStage</td>
+<td>Object containing the id of the program stage.</td>
+<td>"programStage" : {
+"id":"a3kGcGDCuk6"
+}</td>
+</tr>
+<tr class="odd">
+<td>organisationUnit</td>
+<td>Object containing the id of the organisation unit.</td>
+<td>"organisationUnit" : {
+"id":"a3kGcGDCuk6"
+}</td>
+</tr>
+<tr class="even">
+<td>eventQueryCriteria</td>
+<td>Object containing parameters for querying, sorting and filtering events.</td>
+<td>  
+  "eventQueryCriteria": {
+    "status": "COMPLETED",
+    "createdDate": {
+      "from": "2014-05-01",
+      "to": "2019-03-20"
+    },
+    "dataElements": ["a3kGcGDCuk6:EQ:1", "a3kGcGDCuk6"],
+    "filters": ["a3kGcGDCuk6:EQ:1"],
+    "programStatus": "ACTIVE",
+    "ouMode": "SELECTED",
+    "assignedUserMode": "PROVIDED",
+    "assignedUsers" : ["a3kGcGDCuk7", "a3kGcGDCuk8"],
+    "followUp": false,
+    "trackedEntityInstance": "a3kGcGDCuk6",
+    "events": ["a3kGcGDCuk7", "a3kGcGDCuk8"],
+    "fields": "eventDate,dueDate",
+    "order": "dueDate:asc,createdDate:desc"
+  }
+</td>
+</tr>
+</tbody>
+</table>
+
+<table>
+<caption>Event Query Criteria definition</caption>
+<colgroup>
+<col style="width: 33%" />
+<col style="width: 33%" />
+<col style="width: 33%" />
+</colgroup>
+<tbody>
+<tr class="odd">
+<td>followUp</td>
+<td>Used to filter events based on enrollment followUp flag. Possible values are true|false.</td>
+<td>"followUp": true</td>
+</tr>
+<tr class="even">
+<td>ouMode</td>
+<td>To specify the OU selection mode. Possible values are SELECTED| CHILDREN|DESCENDANTS|ACCESSIBLE|CAPTURE|ALL</td>
+<td>"ouMode": "SELECTED"</td>
+</tr>
+<tr class="odd">
+<td>assignedUserMode</td>
+<td>To specify the assigned user selection mode. Possible values are CURRENT| PROVIDED | NONE | ANY </td>
+<td>"assignedUserMode": "NONE"</td>
+</tr>
+<tr class="even">
+<td>assignedUsers</td>
+<td>To specify a list of assigned users for events.</td>
+<td>"assignedUsers": ["a3kGcGDCuk7", "a3kGcGDCuk8"]</td>
+</tr>
+<tr class="odd">
+<td>trackedEntityInstance</td>
+<td>To specify a tracked entity instance that has to be used when filtering events</td>
+<td>"trackedEntityInstance": "A03MvHHogjR"</td>
+</tr>
+<tr class="even">
+<td>order</td>
+<td>To specify ordering/sorting of fields and its directions</td>
+<td>"order"="a3kGcGDCuk6:desc,eventDate:asc"</td>
+</tr>
+<tr class="odd">
+<td>filters</td>
+<td>To specify filters to be applied when listing events</td>
+<td>"filters"=["a3kGcGDCuk6:LT:4"]</td>
+</tr>
+<tr class="even">
+<td>fields</td>
+<td>To specify fields that has to be part of the event listing response.</td>
+<td>"fields"="a3kGcGDCuk6,eventDate,dueDate"</td>
+</tr>
+<tr class="odd">
+<td>dataElements</td>
+<td>To specify dataElements that has to be part of the event grid query response.</td>
```

Comment:
We also need a way of specifying the data value filtering. If you open the capture app you see how the searches will specify filterings for data values in data elements.

Reply:
Shortest reply to all your comments is , I included the FULL capability of our APIs (normal + grid) in this model. Everything that is currently possible in our API as far as filtering/sorting/trimming/anything , its possible to save them in this model. The only thing I missed is the relative periods, which I am working on now. There were some parameters which were exclusively applicable for Grid API and vice versa. I have included both of them in such cases.

Reply:
Ok, so there is a syntax for querying data element values. Could you add an example here perhaps?

Reply:
"dataElement=a3kGcGDCuk6:GT:6" this works for the query end point.